### PR TITLE
Add a /daily plugin

### PR DESCRIPTION
**Pull request by bumbadadabum**

### CODE REVIEW

File: chat-plugins/daily-spotlight.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+let spotlights = {};
+try {
+	spotlights = require(`../${SPOTLIGHT_FILE}`);
+} catch (e) {
+	if (e.code !== 'MODULE_NOT_FOUND' && e.code !== 'ENOENT') throw e;
+}
+if (!spotlights || typeof spotlights !== 'object') spotlights = {};
+
+function saveSpotlights() {
+	FS(SPOTLIGHT_FILE).write(JSON.stringify(spotlights));
```
Comment (Zarel) [inline chat-plugins/daily-spotlight.js]: Why isn't this `writeUpdate`?

File: chat-plugins/daily-spotlight.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+			spotlights[room.id][key].push(obj);
+			this.modlog('QUEUEDAILY', `${key}: ${image ? `${image}, ` : ''}${description}`);
+			this.privateModAction(`${user.name} queued a daily ${key}.`);
+		}
+
+		saveSpotlights();
+	},
+	daily: async function (target, room, user) {
+		let key = toId(target);
+		if (!spotlights[room.id] || !spotlights[room.id][key]) return this.errorReply(`Cannot find a daily spotlight with name '${key}'`);
```
Comment (Zarel) [inline chat-plugins/daily-spotlight.js]:
```suggestion
		if (!spotlights[room.id] || !spotlights[room.id][key]) {
			return this.errorReply(`Cannot find a daily spotlight with name '${key}'`);
		}
```

File: chat-plugins/daily-spotlight.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+		let [key, rest] = target.split(',');
+		key = toId(key);
+		if (!key) return this.parse('/help daily');
+		if (!spotlights[room.id][key]) return this.errorReply(`Cannot find a daily spotlight with name '${key}'`);
+
+		if (!this.can('announce', null, room)) return false;
+		if (rest) {
+			const queueNumber = parseInt(rest);
+			if (isNaN(queueNumber) || queueNumber < 1) return this.errorReply("Invalid queue number");
+			if (queueNumber >= spotlights[room.id][key].length) return this.errorReply(`Queue number needs to be between 1 and ${spotlights[room.id][key].length - 1}`);
```
Comment (Zarel) [inline chat-plugins/daily-spotlight.js]:
```suggestion
			if (queueNumber >= spotlights[room.id][key].length) {
				return this.errorReply(`Queue number needs to be between 1 and ${spotlights[room.id][key].length - 1}`);
			}
```

File: chat-plugins/daily-spotlight.js (hunk 1 above)
Comment (Zarel) [reply chat-plugins/daily-spotlight.js]:
```suggestion
	FS(SPOTLIGHT_FILE).writeUpdate(() => JSON.stringify(spotlights));
```

**Comment by bumbadadabum** [comment]
Man this suggest changes feature is amazing

**Comment by Kaiepi** [comment]
Would `motd`, `setmotd`, etc. make good aliases for this?

**Comment by bumbadadabum** [comment]
I mean they could be, but I personally don't see how "/daily message" and "/motd" are functionally any different.

**Pull request merged**